### DISCUSSION TITLE
Select field for nullable enum

### DISCUSSION THREAD

**Discussion by jonocodes** [aminalaee/sqladmin#773] [General]
Is there a suggested way to get a select field to play nicely with a column that uses a Enum type that is nullable?

Generally I want to use an Enum for possible values. But the column should allow null and the select list should have an empty option. I would like something like this:

```python
class FeeStructuresEnum(Enum):
    unpaid = 'Unpaid'
    paid = 'Paid'

...
Column('fee_structure', ChoiceType(FeeStructuresEnum, impl=sa.String()), nullable=True),
...
```

I have tried these things:

1. Using SelectField and adding various arguments to allow for an empty list item like so:
```python
form_args = {
	fees_structures = dict(choices=..., validators=[Optional()]  default='')
}
```

	This renders the select OK, but does not activate the 'selected' item in the select options.

2. Create my own SelectFieldNullable(SelectField), which behaves as a SelectField but also injects the option ("__None", "") as the QuerySelectField does.

	As above this renders the select OK, but does not activate the 'selected' item in the select options.

3. QuerySelectField has an allow_blank option, which works well when the field is working off an ORM. But when using an Enum type, it gives errors.

```
  File ".../lib/python3.10/site-packages/sqladmin/helpers.py", line 178, in get_primary_keys
    return tuple(inspect(model).mapper.primary_key)
  File ".../lib/python3.10/site-packages/sqlalchemy/inspection.py", line 147, in inspect
    raise exc.NoInspectionAvailable(
sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <enum 'FeeStructuresEnum'>
```

**Comment by mmzeynalli** [comment]
use

```python
form_args = {"fees_structures": {
        "allow_blank": True,
        "blank_text": "-- Choose --",
    }
}
```

as per #1044
